irc.go

```diff
@@ -54,6 +54,89 @@ func (ms *modeSet) Apply(s string) error {
 	return nil
 }
 
+type channelModeType byte
+
+const (
+	modeTypeA channelModeType = iota
```

Comment:
Would be nice to have doc comments about what these are

Reply:
Added.